feat: improve docker installation instructions for clarity and completeness

techmannih opened the pull request:
improve docker installation instructions for clarity and completeness

heliocastro reviewed (changes requested):
@techmannih This need to be reviewed. 
No one need to install hugo or docsy at all, only docker is enough.
Please redo your PR

GMishx commented on README.md:
```diff
@@ -1,78 +1,135 @@
-The website is based on the Hugo static page generator. 
-All relevant source files can be found at github/sw360.website (https://github.com/eclipse/sw360.website). 
-The page is published at eclipse.org/sw360 and build with a jenkins job, which is configured in a jenkins file in the repository of the website.
+<p align="left">
+  <a href="https://github.com/eclipse-sw360/sw360.website/graphs/contributors" alt="SW360 GitHub Contributors">
+    <img src="https://img.shields.io/github/contributors/eclipse-sw360/sw360.website?color=orange" />
+  </a>
+  <a href="https://github.com/eclipse-sw360/sw360.website/issues?q=is%3Aopen" alt="SW360 Website Open Issues">
+    <img src="https://img.shields.io/github/issues/eclipse-sw360/sw360.website?color=%23DDDD22&label=open%20issues" />
+  </a>
+  <a href="https://www.repostatus.org/#active" alt="Repo Status">
+    <img src="https://www.repostatus.org/badges/latest/active.svg" />
+  </a>
+  <a href="https://sw360chat.slack.com/" alt="Join SW360 Slack">
+    <img src="https://img.shields.io/badge/Slack-Join%20SW360-yellow?logo=slack" />
+  </a>
+</p>
```
Don't you think all this makes more sense in the main repo [eclipse-sw360/sw360](https://github.com/eclipse-sw360/sw360) rather than here?

GMishx commented on README.md:
```diff
@@ -110,8 +168,45 @@ sw360.website/
+
+Thanks to all the amazing contributors to this project:
+
+[![Contributors](https://contrib.rocks/image?repo=jeclipse-sw360/sw360.website)](https://github.com/eclipse-sw360/sw360.website/graphs/contributors)
+
+Want to join the list? Check out our [Contributing Guide](CONTRIBUTING.md) to get started!
+
+## Contributing
+
+We’re thrilled to have you join our open-source community! Whether you’re a developer, designer, writer, or enthusiast, your contributions can make a big impact on the SW360 website project. We’ve created a detailed **[CONTRIBUTING.md](CONTRIBUTING.md)** file to guide you, but here’s a quick overview of how you can get involved:
+
+* **Report Bugs & Request Features**
+    Found something broken or have an idea to make the project better? Open an issue on our [GitHub Issues page](https://github.com/eclipse-sw360/sw360.website/issues) with clear details to help us address it quickly.
+
+* **Contribute Code**
+    Want to fix a bug or add a feature? Fork the repository, make your changes, and submit a pull request. Check [CONTRIBUTING.md](CONTRIBUTING.md) for coding guidelines and best practices.
+
+* **Improve Documentation**
+    Clear docs are key to a thriving project. Help us refine setup instructions, add examples, or clarify existing content by submitting updates to this file or others.
+
+* **Enhance the User Interface**
+    Have design skills? Improve the site’s look and feel—whether it’s tweaking the Docsy theme, optimizing layouts, or suggesting accessibility upgrades.
+
+### Connect with the SW360 Community
+
+We’re excited to connect with you! Whether you have questions, feedback, or simply want to chat about the SW360 website, here’s how to reach out and join our community:
+
+* **GitHub Issues**
+    For bug reports, feature requests, or project-related questions, please [open an issue](https://github.com/eclipse-sw360/sw360.website/issues) on our repository. This is the fastest way to get a tracked and timely response!
+* **Slack**
+    Join us on Slack for real-time collaboration, quick questions, or to connect with contributors:
+    <a href="https://sw360chat.slack.com/" target="_blank"><img align="center" src="https://img.icons8.com/color/48/null/slack-new.png" alt="SW360 Slack" height="30" width="40" style="margin-right: 5px;" /></a>
+
+* **Email**
+    For private inquiries, collaboration ideas, or direct feedback, email us at:
+    <a href="mailto:sw360-dev@eclipse.org"><img align="center" src="https://img.icons8.com/color/48/null/gmail.png" alt="SW360 Email" height="30" width="40" style="margin-right: 5px;" /></a>
+
+## License
+
+The contents of this repository are licensed under the Eclipse Public License 2.0 (EPL-2.0). See [LICENSE](./LICENSE).
```
Don't you think all this makes more sense in the main repo [eclipse-sw360/sw360](https://github.com/eclipse-sw360/sw360) rather than here?

GMishx reviewed (changes requested):
Rest of the changes, unfortunately, does not add any value to the documentation.

GMishx commented on README.md:
```diff
@@ -33,72 +33,165 @@ The result of the jenkins build is pushed to: http://git.eclipse.org/c/www.eclip
 
-Ensure you have the following installed before proceeding:
+## Prerequisites
 
-- **Docker** (Check installation by running `docker --version` in the terminal)
-- **Bash** (Available by default on macOS and Linux; for Windows, use Git Bash or WSL)
+Before starting, ensure the following tools are installed on your system:
 
-## Local Development Using Docker
+1. **[Docker Desktop](https://www.docker.com/products/docker-desktop/)**
+    - **Windows/macOS**: Download and install from the official Docker website.
+    - **Linux**: Install via your package manager (e.g., `sudo apt install docker.io` on Ubuntu), then start the service with `sudo systemctl start docker`.
+    - **Verify**: Run `docker --version` to confirm installation.
 
-If you have Docker installed on your system, you can quickly set up a local development environment and test changes in real time.
+2. **Bash Shell**
+    - **macOS/Linux**: Bash is included by default in the terminal.
+    - **Windows**: Use Git Bash (installed with Git) or Windows Subsystem for Linux (WSL).
+    - **Verify**: Run `bash --version` to ensure Bash is available.
 
-## Setting Up the Local Server
+3. **[Go](https://go.dev/doc/install)** (Optional)
+    - Required only for Hugo development or customization (e.g., building from source).
+    - Follow the OS-specific installation instructions on the Go website.
+    - **Verify**: Run `go version` to check the installation.
```
It is not a good idea to document installation guide for other tools as they can change anytime and then you are outdated, showing wrong steps. It should simply say: Setup Docker and here is the installation guide from Docker.

Bash Shell and Go is not required as @heliocastro correctly mentioned, no one should setup this repo manually, bu just us the Docker setup done with `./docker_serve_local.sh` file.

GMishx commented on README.md:
````diff
@@ -33,72 +33,165 @@ The result of the jenkins build is pushed to: http://git.eclipse.org/c/www.eclip
+### Step 1: Clone the Repository
 
-   ```sh
-   http://localhost:1313/sw360
-   ```
+1. Clone the project repository, replacing `{{your_username}}` with your GitHub username:
+    ```bash
+    git clone [https://github.com/](https://github.com/){{your_username}}/sw360.website.git
+    ```
+2. Navigate into the project directory:
+    ```bash
+    cd sw360.website
+    ```
````
Do we really need to spell out each and everything? Anyone here on GitHub should at least know how to fork and clone a repo.

GMishx commented on README.md:
````diff
@@ -33,72 +33,165 @@ The result of the jenkins build is pushed to: http://git.eclipse.org/c/www.eclip
+        bash docker_serve_local.sh
+        ```
+
+3. **Docker Daemon Not Running**
````
Shouldn't it be obvious?

GMishx commented on README.md:
```diff
@@ -33,72 +33,165 @@ The result of the jenkins build is pushed to: http://git.eclipse.org/c/www.eclip
+    3. If the issue persists, restart Docker:
+        - **Linux**: `sudo systemctl restart docker`
+        - **macOS/Windows**: Restart Docker Desktop from the system tray.
+4. **Port Already in Use**
```
Also this?

yashisrani commented:
@GMishx I guess these changes aren’t necessary. The current README instructions are already accurate and well-written.

techmannih commented:
@yashisrani Yeah great, thanks for the feedback! I made the changes based on @GMishx review request. I’ve added separate clarity for both Windows and Linux setup after trying it locally myself. Also included some troubleshooting points that I personally faced during the setup. As for the 'Docker is not running' part, it wasn’t really needed — I’ve updated the setup as per the task requirements and @GMishx suggestions.
